src/ContractInteract/MerklePatriciaProof.js

```diff
@@ -0,0 +1,87 @@
+'use strict';
+
+const Web3 = require('web3');
+
+const AbiBinProvider = require('../AbiBinProvider');
+const Utils = require('../utils/Utils');
+
+const ContractName = 'MerklePatriciaProof';
+
+/**
+ * Contract interact for MerklePatriciaProof.
+ *
+ * As MerklePatriciaProof is a library contract, this only serves for deployment.
+ */
+class MerklePatriciaProof {
+  /**
+   * Constructor for MerklePatriciaProof.
+   *
+   * @param {Object} web3 Web3 object.
+   * @param {string} libraryAddress MerklePatriciaProof contract address.
+   */
+  constructor(web3, libraryAddress) {
+    if (web3 instanceof Web3) {
+      this.web3 = web3;
+    } else {
+      const err = new TypeError(
+        "Mandatory Parameter 'web3' is missing or invalid",
+      );
+      throw err;
+    }
+
+    if (!Web3.utils.isAddress(libraryAddress)) {
+      const err = new TypeError(
+        "Mandatory Parameter 'libraryAddress' is missing or invalid.",
+      );
+      throw err;
+    }
+
+    this.address = libraryAddress;
```

Comment:
in the `constructor`s you are mixing two paradigms:
```
if (good)
  assign to this
else
  fail
```
and
```
if (bad)
  fail

assign to this
```

Would be simpler to have the same paradigm for all fields in the constructors.

Reply:
👍 Should be the early-failure variant everywhere